takwimu/views.py

```diff
@@ -189,12 +192,18 @@ def get(self, request, *args, **kwargs):
             # of profile page
             profilesectionpage_results = ProfileSectionPage.objects.live().search(
```

Comment:
Use the filters to make the search more efficient e.g. Assuming `country_filter` is a list, then:

```python
 ProfileSectionPage.objects.live().filter(parent.name__in=country_filter).search(search_query)
```
I'm not sure about the exact name of `parent.name` but lets make elasticsearch do the work.